### PR TITLE
chores(cmf/sagarouter): jsdoc upgrade

### PR DESCRIPTION
**Pull request by acateland**
**What is the problem this PR is trying to solve?**
no jsdoc available

**What is the chosen solution to this problem?**
added some js doc and a small refactor
http://798.talend.surge.sh/cmf/jsdoc/module-react-cmf_lib_sagaRouter.html

**Please check if the PR fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/cmf/src/sagaRouter/router.js [hunk 1]
```diff
@@ -30,29 +91,39 @@ function shouldRestartSaga(maybeSaga, match) {
...
+ * restart saga if necessary
+ * @param {object} history - react router history
+ * @param {RoutesConfig} routes
+ */
 export default function* sagaRouter(history, routes) {
 	const sagas = {};
 	const routeFragments = Object.keys(routes);
-	while (true) {  // eslint-disable-line no-constant-condition
+	while (true) {
+		// eslint-disable-line no-constant-condition
```
Comment (talend-sromain) [inline packages/cmf/src/sagaRouter/router.js]: i think this linter will not work very well now :)

Comment (acateland) [reply packages/cmf/src/sagaRouter/router.js]: oups

**Comment by jmfrancois** [comment]
you have to add sagaRouter to docs.json

File: packages/cmf/src/sagaRouter/router.js [hunk 2]
```diff
@@ -1,8 +1,55 @@
+/**
+ * @module react-cmf/lib/sagaRouter
+ */
```
Comment (jmfrancois) [inline packages/cmf/src/sagaRouter/router.js]: please add `@example` of usage

File: packages/cmf/src/sagaRouter/router.js [hunk 3]
```diff
@@ -63,7 +133,7 @@ export default function* sagaRouter(history, routes) {
 			}
 			index += 1;
 		}
-		for (let index = 0; index < shouldStart.length;) {
+		for (let index = 0; index < shouldStart.length; ) {
```
Comment (jmfrancois) [inline packages/cmf/src/sagaRouter/router.js]: question: is this a typo ?

**Pull request merged**